chapters/peas-and-pods.rst

```diff
@@ -0,0 +1,122 @@
+TailPea collects the calculated result from the Peas inside the Pod.
+
+Let's say we defined `parallel=3` inside a Pod,
+then we have 5 Peas inside the Pod.
+Traffic first arrives HeadPea, and the HeadPea distributes traffic to the 3 Peas in parallel,
+and finally the result will be sent to the TailPea.
+
+Jina offers parallelization at different levels, such as
+
+1. Local Threads/Process with `BasePea`.
+2. Inside container with `BasePea` inside `ContainerRuntime`.
+3. Remotely with `RemotePeas` and `RemotePods` inside `JinadRuntime`.
+
+Jina Daemon (JinaD) enables Jina to spin up and distribute Peas, Pods, Flows in any system.
+
+Stateless vs Stateful Pea
+---------------------------
+
+In the previous section we mentioned `parallel`.
+This parameter manages different Peas inside a Pod,
+in a Stateless fashion.
+Since a Pea is a wrap over Jina Executor.
+Dependent on the type of the Executor, we consider a Pea is Stateless or Statefull.
+
+If a Pea wraps Encoder, Crafter, Segmenter, Ranker, Evaluator, Classifier, we consider it as Stateless.
+Since this Pea do not persist data.
+On the other hand, if a Pea wraps Indexer, we consider it as Stateful.
+Because we persist indexed result to our workspace.
+
+We use `parallel` to manage the number of Stateless Peas inside a Pod.
+We use `shards` to manage the number of Stateful Peas inside a Pod.
+
+How Local Pea and Remote Pea interact with Pod
+-----------------------------------------------
+
+Let's first think about the case of a Singleton Pod runs locally.
+This Pod will only have 1 Pea inside, i.e. `parallel=1` or `shard=1`,
+depends on the type of Executor it wraps.
+The Pea inside the previous Pod will receive traffic from the last Pea,
+process the data and send data to the next Pea.
+
+When it comes to a Singleton Pod runs remotely (i.e. runs with JinaD),
+the main process will setup `JinadRuntime`,
+the runtime is going to create remote Peas inside the remote machine.
+The JinadRuntime take ownership of 2 tasks:
+
+1. Parse args to JinaD to fire up a remote Pod with a single Pea using `put/pod` api endpoint.
+2. Streaming logs from the remote Pod to the local Pod.
+
+The remote Pea inside the remote Pod will process the data,
+the result will be send to the next Pea.
+In the next section we will talk about the most complex scenario: "distributed Remote Peas in Pod".
+
+Distributed Remote Peas in Pod
+-------------------------------
+
+In this section, we talk about how multiple Peas (`parallel` or `shards` greater or equal to 2) runs inside a remote Pod.
+First of all, the main process will setup `JinadRuntime`.
+`JinadRuntime` create a new remote Pod inside the remote machine, with a HeaderPea,
+a TailPea and `1` Pea inside the same machine,
+the host of the Pea will stay the same as the Pod.
+
+Take a look at the image below.
+Imaging we defined `parallel=2` in our Flow/YAML configuration.
+Since the current remote machine only has 1 ip address (`host=0.0.0.3`),
```

Comment:
has one IP address 
(numbers one to ten should normally be written out, anything more then ten can be a number). 

IP should be uppercase